openfeature/openfeature_api.go

```diff
@@ -6,81 +6,70 @@ import (
 	"sync"
 
 	"github.com/go-logr/logr"
-	"github.com/open-feature/go-sdk/openfeature/internal"
 	"golang.org/x/exp/maps"
 )
 
-// evaluationAPI wraps OpenFeature evaluation API functionalities
-type evaluationAPI struct {
+// ofApiImpl is an internal reference interface extending IOFApi
+type ofApiImpl interface {
+	IOFApi
+	GetProvider() FeatureProvider
+	GetNamedProviders() map[string]FeatureProvider
+	GetHooks() []Hook
+	SetLogger(l logr.Logger)
+	ForEvaluation(clientName string) (FeatureProvider, []Hook, EvaluationContext)
+}
+
+// EvaluationAPI wraps OpenFeature evaluation API functionalities
+type EvaluationAPI struct {
```

Comment:
Is it necessary to export this type as public API? Maybe it would be better to leave it unexported so as not to create undesired dependencies

Reply:
Yup, this can be non-public, will change along with the constructor

Reply:
This is done and I rechecked the same for other structs that should've been internal